immer/algorithm.hpp

```diff
@@ -146,6 +148,23 @@ T accumulate(Iterator first, Iterator last, T init, Fn fn)
     return init;
 }
 
+/*!
+ * Equivalent of the `std::distance` applied to the sentinel-delimited
+ * forward range @f$ [first, last) @f$
+ */
+template <typename Iterator, typename Sentinel,
+          std::enable_if_t
+          <detail::is_forward_iterator<Iterator>
+           and detail::compatible_sentinel<Iterator,Sentinel>, bool> = true>
+std::size_t distance(Iterator first, Sentinel last){
+    std::size_t result = 0;
+    while(first != last){
+        ++first;
+        ++result;
+    }
+    return result;
+}
```

Comment:
I know I should fix this with `clang-format` in the long run, but can you try to keep the style consistent with the rest of the code? In particular in this function:
1. New line between function declaration and opening brace of definition.
2. Space between `if`/`while`/`for` keywords and the opening parens after.
3. AAA (Almost Always use Auto)
4. Prefer `&&`/`||` to `and`.

Reply:
Why is this algorithm added here? For most data-structures there are more efficient ways to compute the distance to a sentinel than just iterating through it. This is particularly true for the data-structures in `immer` so far. I'd rather not provide such a `distance` algorithm or, if provided, provide an optimized version that uses the underlying indexes/structure to be efficient. I can help you if you wanna go in the later direction.

Reply:
```
can you try to keep the style consistent with the rest of the code?
```

Will do!